src/Microsoft.ML.TimeSeries/SrCnnEntireAnomalyDetector.cs

```diff
@@ -359,15 +370,45 @@ public void Train(double[] values, ref double[][] results)
                 {
                     Array.Resize<double[]>(ref results, values.Length);
                 }
-                SpectralResidual(values, results, _threshold);
+
+                bool isTemporal = true;
+                double[] seriesToDetect = values.ToArray();
+
+                if (_period > 0)
+                {
+                    StlConfiguration config = new StlConfiguration(_period);
+                    InnerStl stl = new InnerStl(values, config, isTemporal);
+                    bool success = stl.Decomposition();
+
+                    if (success)
```

Comment:
>success [](start = 24, length = 7)

What happens if `!success`?

Reply:
If not success, just use the original series to do the anomaly detection.

---
In reply to: [435221767](https://github.com/dotnet/machinelearning/pull/5202#discussion_r435221767) [](ancestors = 435221767)